source/js/main.js

```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
+ * 1.add even handler to search button to fetch recipes for query 
+ * 2.add fetched recipes to local Storage
+ * 3.populate recipe cards for search result page
+ * 4.TODO: if no enough result, call getRecipesByAutocomplete
+ */
+ async function searchButtonHandler() {
+  const query = document.querySelector('.form-control').value;
+  const numOfRecipe = 4;    //number of recipe, 2 for testing
+  const pageOffset = 0; //begin with page 0 when search
+
+  const searchResultPageTitle = document.getElementById('search-results-title');
+  searchResultPageTitle.innerHTML = "Top recipes for " + query;
+  const searchResult = await apiFuncs.getRecipesByName(query, numOfRecipe, pageOffset);
+
+  storageFuncs.storeRecipeData(query, searchResult);
+  //API is down, will add Autocomplete once API work
+  // if(searchResult.length < numOfRecipe){
+  //   const numOfAdditionRecipe = numOfRecipe - searchResult.length;
+  //   const autocompleteResult = await apiFuncs.getRecipesByAutocomplete(query, numOfAdditionRecipe);
+  //   storageFuncs.storeRecipeData(query, autocompleteResult);
+  // }
+   
+  const resultRecipeId = JSON.parse(localStorage.getItem('categories'))[query];
+
+  const searchResultCardslocation = document.querySelector('#results-grid-recipe-card-location');
+  removeAllChildNodes(searchResultCardslocation); //refresh cards
+  createRecipeCards(resultRecipeId, searchResultCardslocation, numOfRecipe);
+}
+
+/**
+ * active when show more button click
+ * add # more recipe cards by fetch # more recipe from API with offset
+ * populate more recipe cards
+ */
+async function searchResultShowMore(){
+  const query = document.querySelector('.form-control').value;
+  const numOfCardExist = document.querySelector('#results-grid-recipe-card-location').childElementCount;
+
+  const numOfAdditionRecipeCards = 4; //2 for testing
+  const pageOffset = numOfCardExist/numOfAdditionRecipeCards; //move to next page 
+  
+  const searchResult = await apiFuncs.getRecipesByName(query, numOfAdditionRecipeCards, pageOffset);
+  storageFuncs.storeRecipeData(query, searchResult);
+
+  const searchResultCardslocation = document.querySelector('#results-grid-recipe-card-location');
+  const numOfRecipe = numOfAdditionRecipeCards + numOfCardExist;    //number of total recipe cards
+
+  const NumOfAllRecipeIdInLocal = JSON.parse(localStorage.getItem('categories'))[query].length;
+
+  //create more cards
+  for(let i = numOfCardExist; (i < numOfRecipe) && (i < NumOfAllRecipeIdInLocal); i++){
+    const singleResultRecipeId = JSON.parse(localStorage.getItem('categories'))[query][i];
+    createRecipeCards([singleResultRecipeId], searchResultCardslocation, 1);
+  }
+}
+
 async function init() {
   initializeRoutes();
   bindPopState();
   populateExplore();
   populateSavedRecipes();
   initializeCreateRecipeButtons();
-
+  
   // temporary code
   testCreateRecipeCards();
+  // localStorage.clear(); //for testing, turn off to make less api call
```

Comment:
we have a lot of unnecessary comments everywhere in our code. try to delete things like this before you commit them if we're not planning to keep them

Reply:
roger that